extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -163,15 +163,26 @@ private boolean isHandledByProducers(Type type) {
                 case FLOAT:
                 case LONG:
                 case INT:
+                case SHORT:
+                case BYTE:
+                case CHAR:
                     return true;
                 default:
                     return false;
             }
         }
-        return DotNames.STRING.equals(type.name()) || DotNames.OPTIONAL.equals(type.name()) || SET_NAME.equals(type.name())
-                || LIST_NAME.equals(type.name()) || DotNames.LONG.equals(type.name()) || DotNames.FLOAT.equals(type.name())
-                || DotNames.INTEGER.equals(type.name()) || DotNames.BOOLEAN.equals(type.name())
-                || DotNames.DOUBLE.equals(type.name());
+        return DotNames.STRING.equals(type.name()) ||
+                DotNames.OPTIONAL.equals(type.name()) ||
```

Comment:
I just noticed that `OptionalInt`/`OptionalLong`/`OptionalDouble` are missing from this list.  Might be worth fixing now or later.

Reply:
Sure. I'll do it in a separate PR.